native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java

```diff
@@ -73,6 +74,26 @@ private Decode() {}
 
     public static Object decodeRsaPrivateKeyFromKeyStore(BMap<BString, BString> keyStoreRecord, BString keyAlias,
                                                          BString keyPassword) {
+        Object decodedPrivateKey = getPrivateKey(keyStoreRecord, keyAlias, keyPassword);
+        if (decodedPrivateKey instanceof PrivateKey) {
+            PrivateKey privateKey = (PrivateKey) decodedPrivateKey;
+            return buildRsPrivateKeyRecord(privateKey);
+        }
```

Comment:
I think with java17 we can do this
```suggestion
        if (decodedPrivateKey instanceof PrivateKey privateKey) {
            return buildRsPrivateKeyRecord(privateKey);
        }
```

Reply:
check other places

Reply:
Fixed in [6f1b573](https://github.com/ballerina-platform/module-ballerina-crypto/pull/533/commits/6f1b5734ec97d4e3a5a4e254f44fe11bf72a1202)